src/main/java/org/folio/rest/impl/NotesResourceImpl.java

```diff
@@ -368,149 +358,83 @@ public void getNotesById(String id,
-          okapiHeaders, vertxContext, asyncResultHandler);
-      }
-    });
-  }
-
-  private void putNotesById3Update(String id, String lang, Note entity,
-    Map<String, String> okapiHeaders, Context vertxContext,
-    Handler<AsyncResult<Response>> asyncResultHandler) {
-
-    String tenantId = TenantTool.calculateTenantId(
-      okapiHeaders.get(RestVerticle.OKAPI_HEADER_TENANT));
-    PostgresClient.getInstance(vertxContext.owner(), tenantId)
-      .update(NOTE_TABLE, entity, id, reply -> {
-        if (reply.succeeded()) {
-          if (reply.result().getUpdated() == 0) {
-            asyncResultHandler.handle(succeededFuture(
-              PutNotesByIdResponse.respond500WithTextPlain(
-                messages.getMessage(lang, MessageConsts.NoRecordsUpdated))));
+      getOneNote(id, okapiHeaders, vertxContext)
+        .thenAccept(oldNote -> setNoteCreator(oldNote, note))
+        .thenAccept(o -> setNoteUpdater(note, okapiHeaders))
+        .thenCompose(voidObject -> updateNote(id, lang, note, okapiHeaders, asyncResultHandler, vertxContext))
+        .exceptionally(exception -> {
+          final int cause = ((HttpStatusException) exception.getCause()).getStatusCode();
+
+          if(Response.Status.NOT_FOUND.getStatusCode() == cause) {
+            asyncResultHandler.handle(succeededFuture(PutNotesByIdResponse.respond404WithTextPlain(((HttpStatusException) exception.getCause()).getPayload())));
+          } else if(Response.Status.BAD_REQUEST.getStatusCode() == cause){
+            asyncResultHandler.handle(succeededFuture(PutNotesByIdResponse.respond400WithTextPlain(((HttpStatusException) exception.getCause()).getPayload())));
           } else {
-            asyncResultHandler.handle(succeededFuture(
-              PutNotesByIdResponse.respond204()));
+            asyncResultHandler.handle(succeededFuture(PutNotesByIdResponse.respond500WithTextPlain(exception.getMessage())));
           }
-        } else {
-          ValidationHelper.handleError(reply.cause(), asyncResultHandler);
-        }
+          return null;
       });
   }
 
-  private CompletableFuture<Void> loadTypeNames(List<Note> noteList, Map<String, String> okapiHeaders, Context context, String tenantId) {
+  private void setNoteCreator(Note oldNote, Note note) {
+    final UserDisplayInfo creator = getUserDisplayInfo(oldNote.getCreator().getFirstName(), oldNote.getCreator()
+        .getMiddleName(), oldNote.getCreator().getLastName());
+    note.setCreator(creator);
+  }
+
+  private void setNoteUpdater(Note note, Map<String, String> okapiHeaders) {
+    UserLookUp.getUserInfo(okapiHeaders).thenApply(userLookUp -> getUserDisplayInfo(userLookUp.getFirstName(), userLookUp
+        .getMiddleName(), userLookUp.getLastName())).thenAccept(note::setUpdater);
+  }
+
+  private CompletableFuture<Void> updateNote(String id, String lang, Note note, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
```

Comment:
Looks good @natali-zaitseva -- just a few comments/questions:
- Can we check that `typeId` is valid before updating the note (in case it got changed)
- Should the `type` field be removed (if passed in PUT request) so that it doesn't get written to db

Reply:
Or `type` just needs to be set to readonly in schema definition

Reply:
set `type` as readonly field

Reply:
for `typeId` probably we can create db constraint. will look into this

Reply:
Or could you instead call typeRepository.getTypesByIds to retrieve type name by id (to check it is valid) and to also return type name in the response? Similar to what was done for getNotesById with loadTypeNames

Reply:
db constraint is preferable

Reply:
@dmtkachenko @carolegodfrey added db constraint